src/looprestoration.rs

```diff
@@ -722,44 +721,46 @@ fn selfguided_filter<BD: BitDepth>(
         let mut j = 0;
         while j < h - 1 {
             for i in 0..w {
-                let a = six_neighbors(&B, i as isize);
-                let b = six_neighbors(&A, i as isize);
+                let (a, b) = (six_neighbors(&b, i as isize), six_neighbors(&a, i as isize));
                 dst[i] = ((b - a * src[i].as_::<c_int>() + (1 << 8)) >> 9).as_();
             }
             dst = &mut dst[MAX_RESTORATION_WIDTH..];
             src = &src[REST_UNIT_STRIDE..];
-            B += REST_UNIT_STRIDE;
-            A += REST_UNIT_STRIDE;
+            b += REST_UNIT_STRIDE;
+            a += REST_UNIT_STRIDE;
             for i in 0..w {
-                let a = B[i].as_::<c_int>() * 6 + (B[i as isize - 1] + B[i + 1]).as_::<c_int>() * 5;
-                let b = A[i] * 6 + (A[i as isize - 1] + A[i + 1]) * 5;
+                let (a, b) = (
+                    b[i].as_::<c_int>() * 6 + (b[i as isize - 1] + b[i + 1]).as_::<c_int>() * 5,
+                    a[i] * 6 + (a[i as isize - 1] + a[i + 1]) * 5,
+                );
```

Comment:
Nit for future changes, this wasn't necessary and made the diff harder to review (it's fine to leave here, just makes the review take longer).

Reply:
It was necessary, since `a` and `b` are reassigned.  If you do it one at a time, then the new expression for `b` uses the new definition of `a` instead of the old one.

Reply:
Ahhhh missed that you were using them inside each other now.